### PR TITLE
feat: accept wildcard in matchUpdateTypes array

### PR DESCRIPTION
**Pull request by ferferga**
## Changes:

The ``matchUpdateTypes`` array in config doesn't allow for wildcards, which means some configs get as complex [as this](https://github.com/jellyfin/jellyfin-vue/blob/master/.github/renovate.json#L5)

With this PR, the wildcard character (``*``) is also a documented and valid options, which will add all the possible update types to the array while processing the config.

## Documentation (please check one with an [x])

- [X] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [X] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/config/massage.spec.ts [hunk 1]
```diff
@@ -42,6 +42,18 @@ describe('config/massage', () => {
       expect(res).toMatchSnapshot();
       expect(res.packageRules).toHaveLength(3);
     });
+    it('massages packageRules matchUpdateTypes with wildcard', () => {
+      const config: RenovateConfig = {
+        packageRules: [
+          {
+            // @ts-expect-error - Wildcard is not defined in our types
+            matchUpdateTypes: ['*'],
```
Comment (ferferga) [inline lib/config/massage.spec.ts:50]: I ran the ``npm run prettier-fix && npm run markdown-lint-fix && npm run eslint-fix && npm run doc-fix`` command to fix everything and push changes according to the styles defined for this project, but somehow the automatic linters appended this comma. Not sure if this is wanted here (it's not a very common practice).

Let me know if this is an error so I fix this.

File: lib/config/options/index.ts [hunk 2]
```diff
@@ -1042,6 +1042,7 @@ const options: RenovateOptions[] = [
       'lockFileMaintenance',
       'rollback',
       'bump',
+      '*',
```
Comment (ferferga) [inline lib/config/options/index.ts:1045]: Same as above

File: lib/config/massage.spec.ts (hunk 1 above)
Comment (HonkingGoose) [reply lib/config/massage.spec.ts:50]: Prettier actually switched the default "trailing comma" behavior in the `2.0` release [^docs], quote from their docs:

> ## Trailing commas
> 
> Default value changed from `none` to `es5` in _v2.0.0_
> 
> Print trailing commas wherever possible in multi-line comma-separated syntactic structures. (A single-line array, for example, never gets trailing commas.)
>
> Valid options:
>
> `"es5"` Trailing commas where valid in ES5 (objects, arrays, etc.). No trailing commas in type parameters in TypeScript.

Anyway, we've telling Prettier to use the `es5` style trailing commas via the `.prettierrc` config file:

https://github.com/renovatebot/renovate/blob/cc34a1b4e918b267d6e928b57b31d853fe15223c/.prettierrc#L2

So you can keep the style/formatting as-is. 😉 

[^docs]: https://prettier.io/docs/en/options.html#trailing-commas

**Comment by rarkins** [comment]
I'm not sure I like this implementation approach where it needs to be massaged. Can you open an issue first to describe your requirements and what the implementation approaches can be?

**Comment by ferferga** [comment]
@rarkins Sure, opened at #13317 

Briefly, what I want is to simplify [our long config file at jellyfin-vue due to matchUpdateTypes](https://github.com/jellyfin/jellyfin-vue/blob/master/.github/renovate.json#L2-L48)

**Comment by ferferga** [comment]
Closed as it's unnecessary, as I was able to confirm in #13317

Thank you very much for your feedback and sorry for the inconveniences!

**Pull request closed without merging**